### ISSUE TITLE
Why can we still assign a struct instance's filed when it is already moved ? 

### ISSUE THREAD

**Issue by JerryWang304** [rust-lang/rust#50281]
See the following program.
```Rust
struct Point {
    x: f64,
    y: f64,
}

impl Point {
    fn take_ownership(self) {
        println!("{}", self.x);
    }
}

fn main() {
    let mut p = Point {x:1.0, y:2.0};
    p.take_ownership();
    p.x = 3.0; // Line 15
    println!("{:?}", p.x); // Line 16

```
I wander that why the compiler complains at Line 16 , not Line 15. Both use the moved variable p. I reassign p’s field x at Line 15 and the compiler doesn’t give a warning or an error. For the user’s perspective, Line 15 is useless if so. My question is: what does the compiler do when it comes to Line 15 ?

**Comment by jonas-schievink** [comment]
duplicate of https://github.com/rust-lang/rust/issues/21232

**Comment by Mark-Simulacrum** [comment]
Closing as per previous, duplicate #21232.
